File: src/exemplos/05_EstruturasDeDados/06_Arquivo/05-texto.py
#  -*- coding: utf-8 -*-
#    @package: 05-texto.py
#     @author: Guilherme N. Ramos (gnramos@unb.br)
# @disciplina: Algoritmos e Programação de Computadores
#
# Exemplo de uso de arquivo texto.


if __name__ == '__main__':
    arquivo = 'apc.python.txt'

    try:
        with open(arquivo, 'w') as f:
            f.write('disse o corvo, \'Nunca mais\'.\n')
            f.seek(0)  # Não existe a função 'rewind', mas este é um
                       # comportamento equivalente
            f.write('D')
    except:
        print('Erro!')

File: src/exemplos/05_EstruturasDeDados/06_Arquivo/05-texto.c
/**      @file: 05-texto.c
 *     @author: Guilherme N. Ramos (gnramos@unb.br)
 * @disciplina: Algoritmos e Programação de Computadores
 *
 * Exemplo de uso de arquivo texto. */

#include <stdio.h>
#include <stdlib.h>

int main() {
  char arquivo[] = "apc.c.txt";
  const char string[29] = "disse o corvo, \"Nunca mais\".\n";
  FILE * fp = fopen(arquivo , "w+" );

  if(fp == NULL) {
    printf("Não foi possível abrir \"%s\".\n", arquivo);
    return EXIT_FAILURE;
  }

  fputs(string, fp);
  rewind(fp);  /* <- */
  fputc('D', fp);

  fclose(fp);

  return EXIT_SUCCESS;
}